File: utils/file_manager.py
import logging
import uuid
from pathlib import Path

from config import TEMP_DIR

logger = logging.getLogger(__name__)


def generate_unique_filename(extension: str) -> str:
    unique_id = uuid.uuid4().hex
    return f"{unique_id}.{extension}"


def get_temp_path(filename: str) -> Path:
    return TEMP_DIR / filename


def ensure_temp_dir() -> None:
    TEMP_DIR.mkdir(parents=True, exist_ok=True)
    logger.info(f"Временная директория проверена: {TEMP_DIR}")


def remove_file(file_path: Path) -> bool:
    try:
        if file_path.exists():
            file_path.unlink()
            logger.debug(f"Файл удалён: {file_path}")
            return True
        logger.warning(f"Файл не найден для удаления: {file_path}")
        return False
    except OSError as e:
        logger.error(f"Ошибка при удалении файла {file_path}: {e}")
        return False


def cleanup_temp_files(*file_paths: Path) -> None:
    for file_path in file_paths:
        remove_file(file_path)


def get_file_extension(file_id: str, file_type: str) -> str:
    return "mp4"

File: csharp-bot/Utils/FileManager.cs
using System;
using System.IO;
using BotApp.Config;

namespace BotApp.Utils;

public static class FileManager
{
    public static string GenerateUniqueFilename(string extension)
    {
        var uniqueId = Guid.NewGuid().ToString("N");
        return $"{uniqueId}.{extension}";
    }

    public static string GetTempPath(string filename)
    {
        return Path.Combine(AppConfig.TempDir, filename);
    }

    public static void EnsureTempDir()
    {
        Directory.CreateDirectory(AppConfig.TempDir);
        SimpleLogger.Info($"Временная директория проверена: {AppConfig.TempDir}");
    }

    public static bool RemoveFile(string filePath)
    {
        try
        {
            if (File.Exists(filePath))
            {
                File.Delete(filePath);
                SimpleLogger.Debug($"Файл удалён: {filePath}");
                return true;
            }

            SimpleLogger.Warning($"Файл не найден для удаления: {filePath}");
            return false;
        }
        catch (IOException ex)
        {
            SimpleLogger.Error($"Ошибка при удалении файла {filePath}: {ex}", ex);
            return false;
        }
    }

    public static void CleanupTempFiles(params string[] filePaths)
    {
        foreach (var filePath in filePaths)
        {
            RemoveFile(filePath);
        }
    }
}
